src/DayNine/Smoke2.java

```diff
@@ -27,18 +27,16 @@ public void printLowPointCoordsToHash() {
     }
 
     public void checkBasinSize(int[][] lowPointCoordArray, int[][] heightMap) { //todo
-        Iterator<Point> i = lowPoints.iterator();
-        while (i.hasNext()) {
-            seenPoints.removeAll(seenPoints);
+        for (Point point : lowPoints) {
+            seenPoints.clear();
```

Comment:
Alternatief is `seenPoints=new HashSet<>();` - nieuwe objecten maken is goedkoop.